### ISSUE TITLE
Segfault on character load

### ISSUE THREAD

**Issue by MoonlitRiverlight** [cataclysmbnteam/Cataclysm-BN#1405]
# Describe the bug

Segmentation fault upon loading a save that saved without declared errors.

# Steps To Reproduce

1. Load provided save.

# Expected behavior

Successful load.

# Versions and configuration
- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 21.10; Release: 21.10; Codename: impish; 
- Game Version: f333994 [64-bit]
- Graphics Version: Tiles
- Game Language:  []
- Mods loaded: [
    Bright Nights [dda],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    Elevated bridges [elevated_bridges],
    Arcana and Magic Items [Arcana],
    <color_light_green>Real Dark Skies</color> [realdarkskies],
    <color_light_red>Secronom</color> [secronom],
    <color_light_red>Secronom+ Lore Expansion</color> [secronom_lore_expansion],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    anti-fungal gear [anti_fungal_gears],
    Disinfectant Powder [powder_disinfectant],
    Dried Fruits [dried_fruits],
    Expanded Archery [ExArch],
    Homemade Fungicide [homemade_fungicide],
    MOD to grow tea leaves [More tea leaf],
    The Glory Of Cereal! [CerealMod],
    Sewing Machine [SM],
    Power_Armor_Additions [Power_Armor_Additions],
    More Survivor Notes [more_survive_notes],
    Youkai Enemies [Youkai_Disco],
    Convenient Constructions [convenientconstructions],
    Even More Mutations [even_more_mutations],
    Mythical Martial Arts [MMA],
    Wandering Masters [wandering_masters],
    CDDA Variety Pack [ks_variety_pack],
    Beta National Guard Camp* [national_guard_camp__],
    More Locations* [more_locations__],
    Morlock Village [morlock_village],
    No Rail Stations [No_Rail_Stations],
    Randomized Mall [randomized_mall],
    Trailer Parks [trailerpark],
    Water Towers [water_towers],
    Tankmod: Revived [Tankmod_Revived],
    Magiclysm Revamp and Additions [magicrevamp],
    Graphical Overmap [Graphical_Overmap],
    Hair and Skin Color Cosmetics, UDP and MXotto+ version p10 [mxoudpcosmetics],
    Aftershock [aftershock],
    <color_yellow>Mythical Martial Arts - Mods Extension</color> [MMA_Mods_Extension],
    Autodoc Harmless and Perfect [autodoc_harmless_perfect],
    Manual Bionic Installation [manualbionicinstall],
    Necropolis Actually Findable [necro_find],
    Shinigami [shinigamimod],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Stronger archery [bow_opness],
    Targeted Mutations [TaMu],
    Alternative Map Key [alt_map_key],
    Tame Bugs [tamebugs],
    Miscellaneous Magiclysm Expansions [demon_spider_a],
    Dorf Life [Dorf_Life],
    Folding Parts pack* [deoxymod__],
    sees-player icon, retrodays [sees_player_retro],
    Alternative Basement Layouts [basement_extension]
]

# Additional context

Previously had several crashes on item position when moving any quantity of items and resumed from previous saves.

[crash.log](https://github.com/cataclysmbnteam/Cataclysm-BN/files/8309929/crash.log)
[debug.log](https://github.com/cataclysmbnteam/Cataclysm-BN/files/8309930/debug.log)
[save.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/8309939/save.zip)

**Comment by Coolthulhu** [comment]
In crash.log:
```
game::validate_camps()
```

High chance that #1386 is involved.
Unfortunately faction camp code is very "just so" and doesn't have proper regression testing, so automatic testing didn't find that one.

**Comment by SaintCirno9** [comment]
I think it's not my problem.

Since I reverted #1386 and the error still remains.
![屏幕截图 2022-03-20 122604](https://user-images.githubusercontent.com/33447021/159148247-aaf91e91-b740-4b1e-83df-7e2ce2e65e9c.png)

And this is the error:
![屏幕截图 2022-03-20 122513](https://user-images.githubusercontent.com/33447021/159148273-aed08d62-434e-485f-96ef-480ca8f92dd8.png)

Then I downloaded build 0eaec9d ( Windows tiles x64 untranslated ) to see if the previous version works correctly.
And the same error occurs, build 9941f2b also crashes.
![屏幕截图 2022-03-20 130106](https://user-images.githubusercontent.com/33447021/159148979-ef1218d0-ab35-460f-bbcb-302ec010311c.png)


Sorry for not having the same system environment you use, above results are given by Windows 11 x64, but the error might be
 the same.

Maybe it's the problems of earlier PR? I'm not sure. And what is the previous version you use?

**Comment by Coolthulhu** [comment]
Will require more testing then. I'll try to make a test for this case, maybe it's just lack of checks somewhere.

**Comment by MoonlitRiverlight** [comment]
It happened before attempting to start any faction camp actually. Just started getting segfaults when moving items manually around.
Someone in the cdda boards mentioned a possible fallback for when item positions are lost: https://www.reddit.com/r/cataclysmdda/comments/thy1ty/segmentation_error_on_load/i1asg5f/?context=3

That's quite a bit beyond me, but it doesn't seem like we have that code in BN either.

**Comment by SaintCirno9** [comment]
@MoonlitRiverlight I've found what causes the error.

Have you ever edited your save file manually?
In your debug.log, at the end of it, it says:
```
15:20:37.505 ERROR DEBUGMSG : src/savegame.cpp:253 [void game::unserialize(std::istream&)] Bad save json
Json error: <unknown source file>:1:1: tried to start object, but found '�', not '{'

�
^
 ��# version 28
{
  "turn": 5461176,
```
So, the save is not loaded at all.
And the error, from my experience, is usually a file format problem.
Then I looked into your save (filename `#RGFzaGluYQ==.sav`), and found out it is in format `UTF-8 with BOM`.
A normal save file should be in format `UTF-8`.
After I changed its format to the correct one, the game loaded well.

![屏幕截图 2022-03-22 152136](https://user-images.githubusercontent.com/33447021/159428396-433f08fb-f4e6-42a5-9e31-aa8ac4868bc1.png)

And the character is really powerful!🤣🤣🤣

This is the save file for your convenience.
[save.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/8322044/save.zip)

**Comment by MoonlitRiverlight** [comment]
Yeah, I edited it once it stopped loading! Tried removing active enemies and etc so see if it unbricked the save. I suppose the editor mucked up the format without me realizing it. Was supposed to have returned to the original backup but I assume I got them mixed at some point.

I generated the character with pretty much every starting option I'd mutated/levelled into last time I played. :cake: Thanks for that!
